compiler-core/src/warning.rs

```diff
@@ -382,6 +382,18 @@ expression.",
                         extra_labels: Vec::new(),
                     }),
                 },
+                type_::Warning::DoubleUnary { location } => Diagnostic {
+                    title: "Unnecessary double unary expression".into(),
```

Comment:
This is a bit abstract, which people less familiar with PL jargon will not understand. How about "Pointless double negation" or something like that?

It would also be good to explain to the user why this is a warning. Perhaps something like this:

```
Negating an Int twice returns the original Int, so this expression
does nothing! You can safely remove the two `-`s.
```